port/cpl_float.h

```diff
@@ -567,6 +574,7 @@ template <typename T> constexpr int CPLIsFinite(T x)
     return isfinite(x);
 }
 
+#if !defined(__STDCPP_FLOAT16_T__)
```

Comment:
I notice that you kept the definition of `CPLFloat16`, but are now disabling the specialization for `CPLFloat16`. I would disable the definition of `CPLFloat16` above as well.

Reply:
> I would disable the definition of `CPLFloat16` above as well.

for CPLIsNan(), Inf() ? If I do that , I get
```/home/even/gdal/float16/port/cpl_json_streaming_writer.cpp: In member function 'void CPLJSonStreamingWriter::Add(GFloat16, int)':
/home/even/gdal/float16/port/cpl_json_streaming_writer.cpp:229:9: error: 'CPLIsNan' was not declared in this scope; did you mean 'CPLsscanf'?
  229 |     if (CPLIsNan(hfVal))
```

Reply:
No, I referred to the actual definition of the `class CPLFloat16` and the template specialization for `std::numeric_limits< CPLFloat16>`. These are separated in the source file (because name spaces). I can clean this up while merging.

Reply:
got it. PR adjusted